src/sys/socket/sockopt.rs

```diff
@@ -42,25 +42,33 @@ const TCP_CA_NAME_MAX: usize = 16;
 /// * Type of the value that you are going to set.
 /// * Type that implements the `Set` trait for the type from the previous item (like `SetBool` for
 ///    `bool`, `SetUsize` for `usize`, etc.).
+#[macro_export]
 macro_rules! setsockopt_impl {
     ($name:ident, $level:expr, $flag:path, $ty:ty, $setter:ty) => {
         #[allow(deprecated)] // to allow we have deprecated socket option
-        impl SetSockOpt for $name {
+        impl $crate::sys::socket::SetSockOpt for $name {
             type Val = $ty;
 
-            fn set<F: AsFd>(&self, fd: &F, val: &$ty) -> Result<()> {
-                unsafe {
-                    let setter: $setter = Set::new(val);
-
-                    let res = libc::setsockopt(
+            fn set<F: std::os::unix::io::AsFd>(
+                &self,
+                fd: &F,
+                val: &$ty,
+            ) -> $crate::Result<()> {
+                use $crate::sys::socket::sockopt::Set;
+                let setter: $setter =
+                    $crate::sys::socket::sockopt::Set::new(val);
+                let level = $level;
+                let flag = $flag;
+                let res = unsafe {
+                    libc::setsockopt(
                         fd.as_fd().as_raw_fd(),
```

Comment:
We can import `use std::os::unix::io::AsRawFd` in the scope as well so that users won't need to do this manually:

```
/// Users should be able to define their own sockopts.
mod sockopt_impl {
    use nix::sys::socket::{
        getsockopt, setsockopt, socket, AddressFamily, SockFlag, SockProtocol,
        SockType,
    };
    use std::os::unix::io::AsRawFd; // Users do not need to handle this
```

Reply:
Oh good catch.  I didn't notice that.